File: app/src/main/java/se/hactar/movieregister/helper/imdb/ImdbHelper.java
package se.hactar.movieregister.helper.imdb;


import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import se.hactar.movieregister.helper.imdb.model.Suggest;

public class ImdbHelper {

    public static final String TITLE_URL = "http://www.imdb.com/title/";

    public interface Api {
        String BASE_URL = "http://sg.media-imdb.com/suggests/";

        @GET(BASE_URL + "{firstLetter}/{search}.json")
        Observable<Suggest> getSuggest(@Path("firstLetter") String firstLetter,
                                       @Path("search") String search);
    }
}

File: app/src/main/java/se/hactar/movieregister/helper/imdb/ImdbHelper.kt
package se.hactar.movieregister.helper.imdb


import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path
import se.hactar.movieregister.helper.imdb.model.Suggest

object ImdbHelper {

    const val BASE_URL = "https://sg.media-imdb.com/suggests/"
    const val TITLE_URL = "https://www.imdb.com/title/"

    interface Api {

        @GET("$BASE_URL{firstLetter}/{search}.json")
        fun getSuggest(@Path("firstLetter") firstLetter: String,
                       @Path("search") search: String): Observable<Suggest>
    }
}
